src/main/java/com/surrealdb/refactor/driver/parsing/JsonElementParser.java

```diff
@@ -0,0 +1,61 @@
+package com.surrealdb.refactor.driver.parsing;
+
+import java.net.ProtocolException;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.surrealdb.refactor.exception.UnhandledProtocolResponse;
+import com.surrealdb.refactor.types.QueryResult;
+
+public class JsonElementParser {
```

Comment:
Awesome separating this out, thank you!

Because the purpose of this class it to process results, and the JsonElement type isn't an important detail of the abstraction, I would instead call it something like "ResultParser".

Reply:
sounds good